app/javascript/client/cart/CartContainer.tsx

```diff
@@ -31,6 +31,7 @@ export const CartContainer: FC<CartContainerProps & RouteComponentProps> = ({ hi
     client().watchQuery({ query: GetCartForCartContainerDocument }).subscribe({
         next(data) { refetch() }
     });
+
```

Comment:
Maybe I'm not quite understanding but I thought that `subscribe` would watch the `cart data` and `refetch` if the data changed, causing a re-render in the cart upon `updating cart`

Reply:
Hey, yeah that was my thinking too. But I was really unsure of whether this would work. I feel like maybe we just create a new Context Provider? What do you think? I think it would be pretty simple and is pretty much exactly the kind of use case Context is designed for.